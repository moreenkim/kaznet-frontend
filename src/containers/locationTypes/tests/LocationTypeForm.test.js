// Test LocationTypeForm
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { LocationTypeForm } from '../LocationTypeForm';

describe('containers/locationType/LocationTypeForm', () => {
  it('renders without crashing', () => {
    const initialData = { name: 'Market' };
    shallow(<LocationTypeForm formActionDispatch={function() {}} initialData={initialData} />);
  });

  it('renders locationType form correctly', () => {
    const initialData = { name: 'Market' };
    const wrapper = mount(
      <LocationTypeForm formActionDispatch={function() {}} initialData={initialData} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders locationType form correctly when doing an edit', () => {
    const initialData = {
      name: 'Market'
    };
    const wrapper = mount(
      <LocationTypeForm
        formActionDispatch={function() {}}
        initialData={initialData}
        targetId={1337}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
