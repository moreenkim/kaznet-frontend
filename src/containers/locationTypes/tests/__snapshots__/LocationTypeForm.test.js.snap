// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/locationType/LocationTypeForm renders locationType form correctly 1`] = `
<LocationTypeForm
  formActionDispatch={[Function]}
  initialData={
    Object {
      "name": "Market",
    }
  }
>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "name": "Market",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
  >
    <div>
      <Form
        onSubmit={[Function]}
        tag="form"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <FormGroup
            className="row"
            tag="div"
          >
            <div
              className="row form-group"
            >
              <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-sm-12"
                >
                  <Input
                    aria-label="name"
                    bsSize="lg"
                    className=""
                    name="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Location Type Name"
                    type="text"
                    value="Market"
                  >
                    <input
                      aria-label="name"
                      className="form-control-lg form-control"
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Location Type Name"
                      type="text"
                      value="Market"
                    />
                  </Input>
                </div>
              </Col>
            </div>
          </FormGroup>
          <FormGroup
            className="row mt-5"
            tag="div"
          >
            <div
              className="row mt-5 form-group"
            >
              <Col
                md={
                  Object {
                    "offset": 1,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-5 offset-md-1"
                >
                  <Button
                    className="btn btn-secondary btn-block"
                    color="secondary"
                    onClick={[Function]}
                    tag="button"
                  >
                    <button
                      className="btn btn-secondary btn-block btn btn-secondary"
                      onClick={[Function]}
                      type="button"
                    >
                       
                      Cancel
                       
                    </button>
                  </Button>
                </div>
              </Col>
              <Col
                md={
                  Object {
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-5"
                >
                  <Button
                    className="btn btn-primary btn-block"
                    color="secondary"
                    disabled={false}
                    tag="button"
                    type="submit"
                  >
                    <button
                      className="btn btn-primary btn-block btn btn-secondary"
                      disabled={false}
                      onClick={[Function]}
                      type="submit"
                    >
                      Save Location Type
                    </button>
                  </Button>
                </div>
              </Col>
            </div>
          </FormGroup>
        </form>
      </Form>
      <hr
        className="my-4"
      />
      <Col
        md={
          Object {
            "offset": 4,
            "size": 4,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="col-md-4 offset-md-4"
        >
          <Button
            className="btn btn-secondary btn-block btn-danger center-block"
            color="secondary"
            onClick={[Function]}
            tag="button"
          >
            <button
              className="btn btn-secondary btn-block btn-danger center-block btn btn-secondary"
              onClick={[Function]}
              type="button"
            >
              DELETE LOCATION TYPE
            </button>
          </Button>
        </div>
      </Col>
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
          Object {
            "mountOnEnter": true,
            "timeout": 150,
          }
        }
        centered={false}
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
          Object {
            "timeout": 300,
          }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        role="dialog"
        toggle={[Function]}
        zIndex={1050}
      />
    </div>
  </Formik>
</LocationTypeForm>
`;
